source/simulator/core.cc

```diff
@@ -444,6 +444,15 @@ namespace aspect
       map->initialize(triangulation, MappingQGeneric<dim>(4));
 #endif
 
+    // Check that DG limiters are only used with cartesian mapping
+    if ((parameters.use_limiter_for_discontinuous_temperature_solution ||
+         parameters.use_limiter_for_discontinuous_composition_solution) &&
+        geometry_model->natural_coordinate_system() != Utilities::Coordinates::CoordinateSystem::cartesian)
+      AssertThrow(false,
```

Comment:
Maybe write this as follows so that the condition you're trying to verify shows up in the assertion error message:
```suggestion
    if (parameters.use_limiter_for_discontinuous_temperature_solution ||
        parameters.use_limiter_for_discontinuous_composition_solution)
      AssertThrow(geometry_model->natural_coordinate_system() == Utilities::Coordinates::CoordinateSystem::cartesian,
```